### PR TITLE
Give Detailed information of nn exception

### PR DESCRIPTION
**Pull request by lopelopelope**
## What changes were proposed in this pull request?

Mainly using EL to give detailed information

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ELU.scala [hunk 1]
```diff
@@ -60,7 +60,7 @@ class ELU[T: ClassTag](
 
   override def updateGradInput(input: Tensor[T], gradOutput: Tensor[T]): Tensor[T] = {
     require(input.isSameSizeAs(gradOutput),
-      "input should have the same size with gradOutput")
+      s"input should have the same size with gradOutput ")
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ELU.scala]: why change this line?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala [hunk 2]
```diff
@@ -38,13 +38,15 @@ class Recurrent[T : ClassTag]()
   private var hiddenShape: Array[Int] = null
   private val currentInput = T()
   private val currentGradOutput = T()
-  private val gradInputCell = Tensor[T]()
+  private var gradInputCell = Tensor[T]()
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala]: why change this?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Replicate.scala [hunk 3]
```diff
@@ -38,7 +38,7 @@ class Replicate[T: ClassTag](
   val nDim : Int = Int.MaxValue)
   (implicit ev: TensorNumeric[T]) extends TensorModule[T] {
 
-  require(dim > 0, "Can only replicate across positive integer dimensions.")
+  require(dim > 0, s"Can only replicate across positive integer($dim now) dimensions.")
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Replicate.scala]: what is $dim now

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialConvolution.scala [hunk 4]
```diff
@@ -70,8 +70,8 @@ class SpatialConvolution[T: ClassTag](
   val format: DataFormat = DataFormat.NCHW
 )(implicit ev: TensorNumeric[T]) extends TensorModule[T] with Initializable {
 
-  require(nInputPlane % nGroup == 0, "Number of input channels should be multiples of group.")
-  require(nOutputPlane % nGroup == 0, "Number of output channels should be multiples of group.")
+  require(nInputPlane % nGroup == 0, s"Number of input channels(${nInputPlane} now) should be multiples of group (${nGroup}).")
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialConvolution.scala]: what is ${nInputPlane} now

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialConvolution.scala [hunk 5]
```diff
@@ -70,8 +70,8 @@ class SpatialConvolution[T: ClassTag](
...
+  require(nOutputPlane % nGroup == 0, s"Number of output channels(${nOutputPlane} now) should be multiples of group(${nGroup} now).")
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialConvolution.scala]: (${nOutputPlane} now) ???

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialConvolution.scala [hunk 6]
```diff
@@ -387,7 +387,7 @@ class SpatialConvolution[T: ClassTag](
   }
 
   override def accGradParameters(input: Tensor[T], gradOutput: Tensor[T]): Unit = {
-    require(input.nDimension() == 3 || input.nDimension() == 4, "Only support 3D or 4D input")
+    require(input.nDimension() == 3 || input.nDimension() == 4, s"Only support 3D or 4D input(${input.nDimension()}D now)")
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialConvolution.scala]: what is (${input.nDimension()}D now)

**Comment by yiheng** [comment]
Please run a jenkins job to verify your change

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ELU.scala [hunk 7]
```diff
@@ -59,8 +59,7 @@ class ELU[T: ClassTag](
   }
 
   override def updateGradInput(input: Tensor[T], gradOutput: Tensor[T]): Tensor[T] = {
-    require(input.isSameSizeAs(gradOutput),
-      "input should have the same size with gradOutput")
+    require(input.isSameSizeAs(gradOutput), "input should have the same size with gradOutput ")
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ELU.scala]: it looks like same with before..

**Comment by yiheng** [comment]
@zhichao-li does this change look good to you?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ELU.scala (hunk 7 above)
Comment (zhichao-li) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ELU.scala]: we should log the shape for both ```input``` and ```gradOutput```

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InferReshape.scala [hunk 8]
```diff
@@ -84,7 +84,9 @@ class InferReshape[@specialized(Float, Double) T: ClassTag](
       }
       i += 1
     }
-    require(total <= input.nElement(), "inferred size dim product must be <= total input #elements")
+    require(total <= input.nElement(), "inferred size " +
+      s"dim product must be <= total input #elements" +
+      s"total($total) input(input.nElement)")
```
Comment (zhichao-li) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/InferReshape.scala]: the message should be consistent here. how about ```dim product($total) input(input.nElement)``` ?

**Comment by zhichao-li** [comment]
Maybe we should grep the whole BigDL project one more time as there are still some `require` statement missed here.
i.e: 
```
In View.scala
require(ine % numElements == 0, "input view doesn't match desired view") ```

**Comment by zhichao-li** [comment]
LGTM

**Pull request merged**